### PR TITLE
feat: add tool calling support for sync and async chat completions for mistralai instrumentation

### LINKED ISSUE #312: add function calling support for mistral ai instrumentation

### PR DESCRIPTION
**Pull request by axiomofjoy**
- Adds tool calling support for `mistralai`
- adds tool calling examples for both openai and mistralai
- fixes a mistake with formatting for llm input messages
- bumps up minimum python version to python 3.9

resolves #312

### CODE REVIEW

File: python/instrumentation/openinference-instrumentation-mistralai/src/openinference/instrumentation/mistralai/_chat_wrapper.py [hunk 1]
```diff
@@ -87,17 +95,28 @@ def _get_attributes_from_request(
...
+            logger.exception(
+                f"Failed to get input attributes from request parameters of "
+                f"type {type(request_parameters)}"
+            )
+
+    def _get_extra_attributes_from_request(
+        self,
+        request_parameters: Mapping[str, Any],
+    ) -> Iterator[Tuple[str, AttributeValue]]:
+        # Secondary attributes should be added after input and output to ensure
```
Comment (mikeldking) [inline python/instrumentation/openinference-instrumentation-mistralai/src/openinference/instrumentation/mistralai/_chat_wrapper.py:109]: This is actually not true, attributes fall off the top. There's an issue to track this

**Review by mikeldking — APPROVED** [review]
Rolling🧻

File: python/instrumentation/openinference-instrumentation-mistralai/src/openinference/instrumentation/mistralai/_chat_wrapper.py (hunk 1 above)
Comment (axiomofjoy) [reply python/instrumentation/openinference-instrumentation-mistralai/src/openinference/instrumentation/mistralai/_chat_wrapper.py:109]: So this should actually be reversed?

Comment (axiomofjoy) [reply python/instrumentation/openinference-instrumentation-mistralai/src/openinference/instrumentation/mistralai/_chat_wrapper.py]: addressed in 03f227a1e38e35a10b002e8bd06da1a59801ef5b

Comment (axiomofjoy) [reply python/instrumentation/openinference-instrumentation-mistralai/src/openinference/instrumentation/mistralai/_chat_wrapper.py]: related to #244

Comment (mikeldking) [reply python/instrumentation/openinference-instrumentation-mistralai/src/openinference/instrumentation/mistralai/_chat_wrapper.py]: Thanks!

**Pull request merged**